### PR TITLE
chore(actors): refactor pods.go without actors [DET-9901]

### PR DESCRIPTION
**Pull request by stoksc**
## Description
Change pods.go from an actor to a plain Go struct with locks and make all calls synchronous. The change is simple; most of the work is in making sure this causes no bugs like deadlocks or race conditions, and no performance issues due to lock contention.

## Test Plan
- [x] cover by unit tests and normal "is k8s working" tests

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
DET-9901

### CODE REVIEW

File: master/internal/rm/kubernetesrm/pods.go [hunk 1]
```diff
@@ -615,13 +602,13 @@ func (p *pods) refreshPodStates(ctx *actor.Context, allocationID model.Allocatio
 			continue
 		}
 		pod := pod
-		p.podStatusCallback(ctx.Self().System(), watch.Event{Object: &pod})
+		p.podStatusCallback(system, watch.Event{Object: &pod})
```
Comment (erikwilson) [inline master/internal/rm/kubernetesrm/pods.go:605]: Would it be possible to completely remove actors from pods if we can resolve this? Maybe passing in a callback or interface to Initialize?

**Review by carolinaecalderon — APPROVED** [review]
LGTM

File: master/internal/rm/kubernetesrm/pods.go (hunk 1 above)
Comment (stoksc) [reply master/internal/rm/kubernetesrm/pods.go:605]: I'm going to fix this when i do k8s resource pool. felt it would be easier then.

Comment (stoksc) [reply master/internal/rm/kubernetesrm/pods.go:605]: at that point, i won't be able to ctx.Tell the cluster anyway, so it will have to happen.

**Review by erikwilson — APPROVED** [review]

**Pull request merged**